### PR TITLE
Gsoc webhooks

### PR DESCRIPTION
**Pull request by tgxworld**

### CODE REVIEW

File: spec/jobs/emit_web_hook_event_spec.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+      subject.execute(web_hook_id: topic_hook.id,
+                      event_type: 'topic',
+                      topic_id: topic_with_category.id,
+                      user_id: user.id,
+                      category_id: topic_with_category.category.id)
+    end
+  end
+
+  describe '.web_hook_request' do
```
Comment (tgxworld) [inline spec/jobs/emit_web_hook_event_spec.rb]: @fantasticfears I'm a little concerned with this. Most of the logic of building the request and storing  the response body is not being tested here. Can you see if you can add tests for those?

File: app/models/web_hook.rb [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+  TOPIC_HOOK_3_ARGS = Proc.new do |topic, _, user|
+    WebHook.enqueue_topic_hooks(topic, user)
+  end
+
+  POST_HOOK = Proc.new do |post, _, user|
+    WebHook.enqueue_hooks(:post, post_id: post.id, topic_id: post&.topic&.id, user_id: user&.id, category_id: post.topic&.category&.id)
+  end
+
+  %i(topic_destroyed topic_recovered).each { |event| DiscourseEvent.on(event, &TOPIC_HOOK_2_ARGS) }
```
Comment (tgxworld) [inline app/models/web_hook.rb]: @fantasticfears  I realized that we're missing tests for all this triggers. Can you add them in? :)

**Comment by tgxworld** [comment]
@fantasticfears  I'm doing a final pass before this goes into master! :smile:

**Comment by fantasticfears** [comment]
Sent a pull request to your repo https://github.com/tgxworld/discourse/pull/2

**Pull request merged**